x-pack/plugins/siem/public/store/timeline/helpers.ts

```diff
@@ -1000,6 +1000,13 @@ const removeProvider = (providerId: string, timeline: TimelineModel) => {
   const providerIndex = timeline.dataProviders.findIndex(p => p.id === providerId);
   return [
     ...timeline.dataProviders.slice(0, providerIndex),
+    ...((timeline.dataProviders[providerIndex].and.length && [
```

Comment:
optional nit: a ternary operator `?` in lieu of the `&&` eliminates the extra `(` `)`, and might be (ever so) slightly more semantic:

```
const removeProvider = (providerId: string, timeline: TimelineModel) => {
  const providerIndex = timeline.dataProviders.findIndex(p => p.id === providerId);
  return [
    ...timeline.dataProviders.slice(0, providerIndex),
    ...(timeline.dataProviders[providerIndex].and.length
      ? [
          {
            ...timeline.dataProviders[providerIndex].and.slice(0, 1)[0],
            and: [...timeline.dataProviders[providerIndex].and.slice(1)],
          },
        ]
      : []),
    ...timeline.dataProviders.slice(providerIndex + 1),
  ];
};
```

Reply:
I started out with a ternary but at some point it got moved over to an `&&` before I got it just right. Totally agree on readability. Incoming! 🚀